pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -407,9 +424,9 @@ stdenv.mkDerivation {
 
   setupHooks = [
     ../setup-hooks/role.bash
-  ] ++ lib.optional (cc.langC or true) ./setup-hook.sh
-    ++ lib.optional (cc.langFortran or false) ./fortran-hook.sh
-    ++ lib.optional (targetPlatform.isWindows) (stdenv.mkDerivation {
+  ] ++ optional (cc.langC or true) ./setup-hook.sh
+    ++ optional (cc.langFortran or false) ./fortran-hook.sh
+    ++ optional (targetPlatform.isWindows) (stdenv.mkDerivation {
```

Comment:
As a side quest, it would be nice to clarify the usage of `optional` versus `optionals`. Here's a very good example.

My perspective leans towards favouring `optionals` for a couple of reasons. Given that `setupHooks` is inherently a list, using `optional` can misleadingly suggest that adding a single item, such as `./setup-hook.sh`, directly to it is possible in Nix (_spoiler: it's not, only a list can be added to another list_).

In contrast, `optional` implicitly wraps elements into a list, which could inadvertently complicate the learning process for new users of the ecosystem. Opting for `optionals` might alleviate some of this confusion by better indicating that multiple items can be conditionally added.

Reply:
Can you phrase this as an (documentation?) issue so we have something to hold on to? Once this is merged no one will read this comment again.